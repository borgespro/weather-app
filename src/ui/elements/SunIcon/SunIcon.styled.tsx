import styled from 'styled-components/native';

import { CustomThemeProps } from '@/typings/theme';

export const SunIconContainer = styled.View`
  justify-content: center;
  padding-right: ${(props: CustomThemeProps) => props.theme.spacing.xxs};
`;
export const SunIconImg = styled.Image`
  width: ${(props: CustomThemeProps) => props.theme.spacing.xl};
  height: ${(props: CustomThemeProps) => props.theme.spacing.xl};
`;
