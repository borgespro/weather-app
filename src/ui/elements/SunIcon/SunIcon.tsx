import React from 'react';

import { SunIconContainer, SunIconImg } from '@/ui/elements/SunIcon/SunIcon.styled';

export default function SunIcon(): React.ReactElement {
  return (
    <SunIconContainer>
      <SunIconImg source={require('@/assets/sun.png')} />
    </SunIconContainer>
  );
}
