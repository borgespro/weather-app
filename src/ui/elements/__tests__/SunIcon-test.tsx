import React from 'react';

import { shallow } from 'enzyme';
import toJson from 'enzyme-to-json';

import SunIcon from '@/ui/elements/SunIcon';

describe('SunIcon Element', () => {
  it('should render without issues', () => {
    const component = shallow(<SunIcon />);
    expect(component.length).toBe(1);
    expect(toJson(component)).toMatchSnapshot();
  });
});
