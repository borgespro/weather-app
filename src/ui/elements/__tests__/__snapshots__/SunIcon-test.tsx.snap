// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SunIcon Element should render without issues 1`] = `
<Styled(View)>
  <Styled(Image)
    source={1}
  />
</Styled(View)>
`;
